### PR TITLE
create external cluster machine deployment

### LINKED ISSUE #8309: create external cluster machine deployment
**User Story**

create external cluster machine deployment

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: create external cluster machine deployment. Endpoint creates a new node pool for the GKE provider. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #8309

```release-note
NONE
```

### CODE REVIEW

File: pkg/handler/v2/external_cluster/gke.go [hunk 1]
```diff
@@ -271,3 +298,66 @@ func getGKEMachineDeployment(ctx context.Context, svc *container.Service, projec
...
+			Autoprovisioned: gke.Autoscaling.Autoprovisioned,
+			Enabled:         gke.Autoscaling.Enabled,
+			MaxNodeCount:    gke.Autoscaling.MaxNodeCount,
+			MinNodeCount:    gke.Autoscaling.MinNodeCount,
+		}
+	}
+	if gke.Management != nil {
+		nodePool.Management = &container.NodeManagement{
+			AutoRepair:  gke.Management.AutoRepair,
+			AutoUpgrade: gke.Management.AutoRepair,
```
Comment (lsviben) [inline pkg/handler/v2/external_cluster/gke.go]:
```suggestion
			AutoUpgrade: gke.Management.AutoUpgrade,
```

Comment (zreigz) [reply pkg/handler/v2/external_cluster/gke.go]: done

**Pull request merged**